### PR TITLE
Memepad

### PR DESCRIPTION
**Pull request by jvonasek**

### CODE REVIEW

File: src/api/balances.ts [hunk 1]
```diff
@@ -56,15 +56,16 @@ export const getTokenBalance =
 export const useTokenBalance = (
   id: Maybe<string | u32>,
   address: Maybe<AccountId32 | string>,
+  options: { refetchInterval?: number } = {},
```
Comment (vkulinich-cl) [inline src/api/balances.ts:59]: Shouldn't options be an optional argument?

File: src/api/xcm.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  chains: chainsMap,
+  chainsConfig: chainsConfigMap,
+})
+
+export const wallet = new Wallet({
+  config: xcmConfigService,
+})
+
+export const createXcmAssetKey = (id: string, symbol: string) => {
+  if (id && symbol) {
```
Comment (vkulinich-cl) [inline src/api/xcm.ts:33]: why is it here since id an symbol are required arguments?

File: src/api/balances.ts (hunk 1 above)
Comment (jvonasek) [reply src/api/balances.ts]: removed the options because I dont need anymore

File: src/api/xcm.ts (hunk 2 above)
Comment (jvonasek) [reply src/api/xcm.ts]: fixed

File: src/sections/memepad/components/MemepadSummary/MemepadSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    symbol,
+    name,
+    decimals,
+    account,
+    supply,
+    xykPoolAssetId,
+    internalId,
+  } = values || {}
+
+  const xykAssetAMeta = internalId ? assets.getAsset?.(internalId) : null
```
[the remark this answers is not in the document]
Comment (jvonasek) [reply src/sections/memepad/components/MemepadSummary/MemepadSummary.tsx]: fixed

File: src/sections/memepad/form/MemepadForm.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { Stepper } from "components/Stepper/Stepper"
+import { useTranslation } from "react-i18next"
+import { MemepadSpinner } from "sections/memepad/components/MemepadSpinner"
+import { useMemepadFormContext } from "./MemepadFormContext"
+import { useInterval } from "react-use"
+import { useState } from "react"
+
+const MEME_TEXTS = [
```
[the remark this answers is not in the document]
Comment (jvonasek) [reply src/sections/memepad/form/MemepadForm.tsx]: fixed

File: src/sections/memepad/form/MemepadFormFields.tsx [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+                error={
+                  isDirty && !!form.formState.touchedFields?.supply
+                    ? error?.message
+                    : undefined
+                }
+              />
+            )}
+          />
+          <MemepadTokenPrice
+            label="Initial token price"
```
[the remark this answers is not in the document]
Comment (jvonasek) [reply src/sections/memepad/form/MemepadFormFields.tsx]: fixed

**Review by vkulinich-cl — APPROVED** [review]

**Pull request merged**